portal/src/ReactApp.tsx

```diff
@@ -205,6 +210,35 @@ const defaultComponents = {
   DocLink,
 };
 
+export interface LoadCurrentUserProps {
+  children?: React.ReactNode;
+}
+
+const LoadCurrentUser: React.FC<LoadCurrentUserProps> =
+  function LoadCurrentUser({ children }: LoadCurrentUserProps) {
+    const { viewer } = useViewerQuery();
+
+    const gtmEvent = useAuthgearGTMEvent({
+      event: AuthgearGTMEventType.Identified,
+    });
+    const sendDataToGTM = useGTMDispatch();
+    useEffect(() => {
+      if (viewer) {
+        const event: AuthgearGTMEvent = {
+          ...gtmEvent,
+          event_data: {
+            // fixme: userid encoded twice
+            user_id: extractRawID(extractRawID(viewer.id)),
+            email: viewer.email ?? undefined,
+          },
+        };
+        sendDataToGTM(event);
+      }
+    }, [viewer, gtmEvent, sendDataToGTM]);
```

Comment:
My suggestion.

We only handle user identification here. The reason is that this place is the most accurate place of knowing which account is signed in. We always fire the lifecycle event here. In the custom HTML, we always call reset(). If the viewer is present, we call identify(). The only downside is that refresh the portal will create a new mixpanel session, but the mixpanel session will always be identified correctly.